tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py

```diff
@@ -0,0 +1,216 @@
+        # ('v0.10.0', 'gcr.io/grpc-testing/td-grpc-bootstrap:66de7ea0e170351c9fae17232b81adbfb3e80ec3'),
+    )
+    def test_baseline_in_client_with_bootstrap_version(self, image):
+        """Runs the baseline test for multiple versions of the bootstrap
+        generator on the client.
+        """
+        self.client_runner = self.initKubernetesClientRunner(
+            td_bootstrap_image=image)
+        test_client: XdsTestClient = self.startTestClient(self.test_server)
+        self.assertXdsConfigExists(test_client)
+        self.assertSuccessfulRpcs(test_client)
+
+
+# TODO: Use unique client and server deployment names while creating the
+# corresponding runners, by suffixing the version of the bootstrap generator
+# being tested. Then, run these in parallel.
+class BootstrapGeneratorServerTest(
+        bootstrap_generator_testcase.BootstrapGeneratorBaseTest,
+        parameterized.TestCase):
+    server_runner: KubernetesServerRunner
+    client_runner: KubernetesClientRunner
+    test_server: XdsTestServer
+
+    def tearDown(self):
+        logger.info('----- TestMethod %s teardown -----', self.id())
+        retryer = retryers.constant_retryer(wait_fixed=_timedelta(seconds=10),
+                                            attempts=3,
+                                            log_level=logging.INFO)
+        try:
+            retryer(self._cleanup)
+        except retryers.RetryError:
+            logger.exception('Got error during teardown')
+        super().tearDown()
+
+    def _cleanup(self):
+        self.client_runner.cleanup(force=self.force_cleanup)
+        self.removeServerBackends()
+        self.server_runner.cleanup(force=self.force_cleanup)
+
+    @parameterized.named_parameters(
+        # Add images corresponding to future releases here.
+        #
+        # TODO: Update bootstrap generator release instructions to add a newly
+        # released version to this list.
+        ('v0.14.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:d6baaf7b0e0c63054ac4d9bedc09021ff261d599'
+        ),
+        ('v0.13.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:203db6ce70452996f4183c30dd4c5ecaada168b0'
+        ),
+        ('v0.12.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:8765051ef3b742bc5cd20f16de078ae7547f2ba2'
+        ),
+        ('v0.11.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:b96f7a73314668aee83cbf86ab1e40135a0542fc'
+        ),
+        # v0.10.0 uses v2 xDS transport protocol by default. TD only supports v3
+        # and we can force the bootstrap generator to emit config with v3
+        # support by setting the --include-v3-features-experimental flag to
+        # true.
+        #
+        # TODO: Figure out how to pass flags to the bootstrap generator via the
+        # client and server runners, and uncomment this version.
+        # ('v0.10.0', 'gcr.io/grpc-testing/td-grpc-bootstrap:66de7ea0e170351c9fae17232b81adbfb3e80ec3'),
```

Comment:
It will be amazing if you could move this block, and its copy above to a method. You won't have to repeat it then. Take a look at this [absl usage example](https://github.com/abseil/abseil-py/blob/58ead8c22230a2493006fa0ab9f76776b6e7280f/absl/testing/parameterized.py#L117-L128):

```
Inputs from Iterables
=====================
If parameters should be shared across several test cases, or are dynamically
created from other sources, a single non-tuple iterable can be passed into
the decorator. This iterable will be used to obtain the test cases:

  class AdditionExample(parameterized.TestCase):
    @parameterized.parameters(
      c.op1, c.op2, c.result for c in testcases
    )
    def testAddition(self, op1, op2, result):
      self.assertEqual(result, op1 + op2)
```

I think it'll be something like (just an illustration, didn't test the code).

This is just defined at the top level in this file:

```py
def bootstrap_version_testcases() -> List:
  return (
        ('v0.14.0',
         'gcr.io/grpc-testing/td-grpc-bootstrap:d6baaf7b0e0c63054ac4d9bedc09021ff261d599'
        ),
        ('v0.13.0',
         'gcr.io/grpc-testing/td-grpc-bootstrap:203db6ce70452996f4183c30dd4c5ecaada168b0'
        ),
        # ...
    )
```

and then parametrized could be 


```py
@parameterized.parameters(bootstrap_version_testcases())
def test_baseline_in_server_with_bootstrap_version(self, image):
```

Reply:
Done. Thanks.